lib/EmitterInterface.php

```diff
@@ -47,7 +47,7 @@ public function once(string $eventName, callable $callBack, int $priority = 100)
      * Lastly, if there are 5 event handlers for an event. The continueCallback
      * will be called at most 4 times.
      *
-     * @param array<int, mixed>    $arguments
+     * @param list<mixed>          $arguments
```

Comment:
not sure this will work with named arguments. do we have a test-case for that?

Reply:
If I change to `array<int|string, mixed>` then phpstan complains:
```
------ ---------------------------------------------------------------------------------------------------------------------- 
  Line   lib/EmitterTrait.php (in context of class Sabre\Event\Emitter)                                                        
 ------ ---------------------------------------------------------------------------------------------------------------------- 
  80     Parameter #2 $parameters of function call_user_func_array expects array<int, mixed>, array<int|string, mixed> given.  
  91     Parameter #2 $parameters of function call_user_func_array expects array<int, mixed>, array<int|string, mixed> given.  
 ------ ---------------------------------------------------------------------------------------------------------------------- 
```

phpstan thinks that the PHP function `call_user_func_array` must be passed just a "list" array (int keys).

But https://w ww.php.net/manual/en/function.call-user-func-array.php says:
"If any keys of args are strings, those elements will be passed to callback as named arguments, with the name given by the key."

Maybe phpstan needs to be adjusted so that it knows what can be passed to `call_user_func_array` ?

Reply:
> not sure this will work with named arguments. do we have a test-case for that?

The tests only have cases with a single unnamed argument. Named arguments only became a thing in PHP 8.0 (I think). So that will be why there are no test cases for it.

Hmmm - so maybe the phpstan thing in comment above is because I am running phpstan on PHP 7.4.

I think that the code will "just work". But annotations and test case(s) will have to apply only for PHP 8.
Should we try to add official support for named arguments now?

Reply:
I remember working on this topic in https://github.com/phpstan/phpstan-src/pull/2501/ months ago

its a open issue: https://github.com/phpstan/phpstan/issues/5934